src/dmd/expressionsem.d

```diff
@@ -6228,6 +6229,37 @@ private extern (C++) final class ExpressionSemanticVisitor : Visitor
                 }
             }
         }
+        /* https://issues.dlang.org/show_bug.cgi?id=809
+         *
+         * If the address of a lazy variable is taken,
+         * the expression is rewritten so that the type
+         * of it is the delegate type. This means that
+         * the symbol is not going to represent a call
+         * to the delegate anymore, but rather, the
+         * actual symbol.
+         */
+        if (exp.e1.op == TOK.variable)
+        {
+            VarExp ve = cast(VarExp)exp.e1;
+            if (ve.var.storage_class & STC.lazy_)
+            {
+                exp.e1 = exp.e1.expressionSemantic(sc);
+                exp.e1 = resolveProperties(sc, exp.e1);
+                if (exp.e1.op == TOK.call)
+                {
+                    auto callExp = cast(CallExp)exp.e1;
+                    if (callExp.e1.type.ty == Tdelegate)
+                    {
+                        VarExp ve2 = cast(VarExp)callExp.e1;
```

Comment:
Use `isVarExp()`. Also, how do you know callExp.e1 is a VarExp?

Reply:
Because resolvePropertiesX replaces the VarExp with a CallExp for which e1 is the initial VarExp. (look at line 1302 in expressionsem.d)